src/state/mfe/actions.js

```diff
@@ -35,3 +35,19 @@ export const fetchMfeConfigList = (page = { page: 1, pageSize: 0 }, params = '')
         dispatch(addToast(error.message, TOAST_ERROR));
       });
   });
+
+// creating a new action as we don't want to dispatch error toasts on MFeDownloadManager
+export const fetchInitialMfeConfigList = (page = { page: 1, pageSize: 0 }, params = '') => dispatch =>
+  new Promise((resolve, reject) => {
+    getMfeConfigList(page, params)
+      .then((response) => {
+        response.json().then((json) => {
+          if (response.ok) {
+            dispatch(setMfeConfigList(json.payload));
+            resolve(response);
+          } else {
+            reject();
+          }
+        }).catch(() => reject());
+      }).catch(() => reject());
+  });
```

Comment:
instead of duplicating the action, why don't add a boolean param `withToastNotification`  to `fetchMfeConfigList` to disable the toasts notification? the logic is the same.

Reply:
the implementation is slightly different as well.. I also change the way we resolve/reject the promises, I guess adding if/else statement would turn the method very difficult to read.

this is how it would look like:

```js
export const fetchMfeConfigList = (page = { page: 1, pageSize: 0 }, params = '', withToastNotification) => dispatch =>
  new Promise((resolve, reject) => {
    getMfeConfigList(page, params)
      .then((response) => {
        response.json().then((json) => {
          if (response.ok) {
            dispatch(setMfeConfigList(json.payload));
            if (withToastNotification) {
              resolve(response);
            }
          } else if (withToastNotification) {
            json.errors.forEach(err => dispatch(addToast(err.message, TOAST_ERROR)));
          } else {
            reject();
          }
        }).catch((error) => {
          if (withToastNotification) {
            dispatch(addToast(error.message, TOAST_ERROR));
          } else {
            reject();
          }
        });
      }).catch((error) => {
        if (withToastNotification) {
          dispatch(addToast(error.message, TOAST_ERROR));
        } else {
          reject();
        }
      });
  });
```

Reply:
If you want to improve the readability you could use a functional approach, for example:

```javascript
function notify({message, enabled}, cb) {
    if (enabled) return dispatch(addToast(message, TOAST_ERROR));

    return cb ? cb() : null
}

export const fetchMfeConfigList = (page = {page: 1, pageSize: 0}, params = '', withToastNotification) => dispatch =>
    new Promise((resolve, reject) => {
        getMfeConfigList(page, params)
            .then((response) => {
                response.json().then((json) => {
                    if (response.ok) {
                        dispatch(setMfeConfigList(json.payload));
                        resolve(response);
                    } else if (withToastNotification) {
                        json.errors.forEach(err => notify({ message: err.message, enabled: true }));
                    }

                    reject()
                }).catch((error) => {
                   notify({ message: error.message, enabled: withToastNotification }, reject)
                });
            }).catch((error) => {
                notify({ message: error.message, enabled: withToastNotification }, reject)
            });
    });
```

Regarding the `resolve` and `reject`, I think you could normalize also. For what I see `fetchMfeConfigList` it's not used with handlers on `then` and `catch` so we could set the behavior without a problem.